lib/rubocop/cop/lint/redundant_safe_navigation.rb

```diff
@@ -81,15 +85,47 @@ module Lint
       #   do_something if attrs.nil_safe_method(:[])
       #   do_something if attrs&.not_nil_safe_method(:[])
       #
+      # @example CheckReceiverPreviousOccurrences: false (default)
+      #   # good
+      #   foo.bar
+      #   foo&.baz
+      #
+      # @example CheckReceiverPreviousOccurrences: true
+      #   # bad
+      #   foo.bar
+      #   foo&.baz # would raise on previous line if `foo` is nil
+      #
+      #   # good
+      #   foo.bar
+      #   foo.baz
+      #
+      #   # bad
+      #   if foo.condition?
+      #     foo&.bar
+      #   end
+      #
+      #   # good
+      #   if foo.condition?
+      #     foo.bar
+      #   end
+      #
       class RedundantSafeNavigation < Base
         include AllowedMethods
         extend AutoCorrector
 
         MSG = 'Redundant safe navigation detected, use `.` instead.'
         MSG_LITERAL = 'Redundant safe navigation with default literal detected.'
+        MSG_NON_NIL =
+          'Redundant safe navigation on non nil receiver (detected by previous code paths).'
 
         NIL_SPECIFIC_METHODS = (nil.methods - Object.new.methods).to_set.freeze
 
+        NIL_METHODS = (
+          nil.methods +
+          %i[present? blank? try try!] + # rails methods
```

Comment:
Probably non-core methods should be in a configuration option. I'm guessing some people will want to add more methods here anyways for whatever reasons.